modules/gateway/peers.go

```diff
@@ -39,6 +41,7 @@ func (g *Gateway) removePeer(peer modules.NetAddress) error {
 	}
 	delete(g.peers, peer)
 	g.save()
+	g.log.Println("INFO: removed peer", peer)
```

Comment:
Is there a way to get a reason for the peer's removal? I know there are strikes and there are Remove calls, perhaps this log should be moved to the caller instead of inside `removePeer`

I feel like logging should be as top level as possible, because it gives you more context.